### PR TITLE
feat : add update case study handler to update users and engagements

### PR DESCRIPTION
**Pull request by rishusingh022**

### CODE REVIEW

File: src/services/case-study.service.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+  // update the user table
+  if (deletedCaseStudy.collaboratorsIds) {
+    let collaborators = deletedCaseStudy.collaboratorsIds;
+    await userServices.removeCaseStudyFromUser(collaborators, id);
+  }
+  // update the project table
+  if (deletedCaseStudy.engagementId) {
+    let engagement = deletedCaseStudy.engagementId;
+    await projectServices.removeCaseStudyFromProject(engagement, id);
+  }
```
Comment (rohini-senthil) [inline src/services/case-study.service.js:19]: The calls made to other services should be moved to the controller level. 

```
deleteController () => {
  // update users
 // update engagement
 // delete case study
} 
```

Comment (rishusingh022) [reply src/services/case-study.service.js:19]: okay sure !!

Comment (rishusingh022) [reply src/services/case-study.service.js:19]: but  i am accessing old and new collaborator as a parameter to services then i have to pass old collobator old engagement new collaborator new engagement and updated case study as a result to controller function should i do that ?

File: tests/__mocks__/case-study.js [hunk 2]
```diff
@@ -14,15 +14,17 @@ const update = {
...
+    caseStudy: {
+      case_study_id: '1',
+      name: 'test',
+      description: 'test',
+      collaborators_ids: ['1', '2'],
+      image: 'test',
+      box_link: 'test',
+      engagement_id: '1',
+      createdAt: '2022-01-17T04:33:12.000Z',
+      updatedAt: '2022-01-17T04:33:12.000Z',
```
Comment (rohini-senthil) [inline tests/__mocks__/case-study.js:26]: Can we use consistent casing?

File: src/services/project.service.js [hunk 3]
```diff
@@ -45,6 +45,43 @@ const deleteProject = async projectId => {
   });
 };
 
+const updateCaseStudyInProject = async (oldEngagement, newEngagement, caseStudyId) => {
```
Comment (rohini-senthil) [inline src/services/project.service.js:48]: Lets also include the respective unit tests for any new service functions added

File: src/controllers/case-study.controller.js [hunk 4]
```diff
@@ -1,19 +1,30 @@
...
+const projectServices = require('../services/project.service');
 const updateCaseStudy = async (req, res) => {
   try {
     logger.info('updating casestudy with id: ' + req.params.id);
     const { id } = req.params;
     const { body } = req;
 
-    const updatedCaseStudy = await caseStudyServices.updateCaseStudy(id, body);
+    const { caseStudy, oldCollaborators, newCollaborators, oldEngagement, newEngagement } =
+      await caseStudyServices.updateCaseStudy(id, body);
```
[the remark this answers is not in the document]
Comment (rishusingh022) [reply src/controllers/case-study.controller.js:12]: Okay

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,4 @@
 node_modules
 .DS_Store
 .env
-coverage
```
[the remark this answers is not in the document]
Comment (rishusingh022) [reply .gitignore:4]: okay

[the remark this answers is not in the document]
Comment (rishusingh022) [reply .gitignore:4]: i need to remove the coverage ?

File: database/config/config.json [hunk 6]
```diff
@@ -2,10 +2,10 @@
   "development": {
     "username": "postgres",
     "password": null,
-    "database": "staffing_pipeline_db",
```
[the remark this answers is not in the document]
Comment (rishusingh022) [reply database/config/config.json:5]: sure !

**Review by rohini-senthil — APPROVED** [review]

**Pull request merged**